storage/replica_command.go

```diff
@@ -467,6 +467,20 @@ func (r *Replica) BeginTransaction(
 		}
 	}
 
+	r.mu.Lock()
+	threshold := r.mu.state.TxnSpanThreshold
+	r.mu.Unlock()
+
+	// Disallow a transaction record to be created if it's at a timestamp below
```

Comment:
How about "Disallow creation of transaction record if it's at a timestamp before..."